compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/CupertionTextFieldPointerModifier.skiko.kt

```diff
@@ -132,24 +133,86 @@ private fun getTapHandlerModifier(
     }
 }
 
+/**
+ * Returns a modifier which allows to precisely move the caret in the text by drag gesture after long press
+ *
+ * @param state The state of the text field.
+ * @param offsetMapping The offset mapping of the text field.
+ * @return A modifier that handles long press and drag gestures.
+ */
 @Composable
-private fun getSelectionModifier(manager: TextFieldSelectionManager): Modifier {
-    val currentManager by rememberUpdatedState(manager)
+private fun getLongPressHandlerModifier(
+    state: TextFieldState,
+    offsetMapping: OffsetMapping
+): Modifier {
+    val currentState by rememberUpdatedState(state)
+    val currentOffsetMapping by rememberUpdatedState(offsetMapping)
+    var dragTotalDistance = remember { Offset.Zero }
+    var dragBeginOffset = remember { Offset.Zero }
+    var currentDragPosition: Offset?
+
+    val longTapActionsObserver = object : TextDragObserver {
+        // Unnecessary
+        override fun onDown(point: Offset) {}
+
+        // Unnecessary
+        override fun onUp() {}
+
+        override fun onStart(startPoint: Offset) {
+            currentState.layoutResult?.let { layoutResult ->
+                TextFieldDelegate.setCursorOffset(
+                    startPoint,
+                    layoutResult,
+                    currentState.processor,
+                    currentOffsetMapping,
+                    currentState.onValueChange
+                )
+                dragBeginOffset = startPoint
+                currentDragPosition = dragBeginOffset
+            }
+            dragTotalDistance = Offset.Zero
+            currentState.showFloatingToolbar = false
+        }
+
+        override fun onDrag(delta: Offset) {
+            dragTotalDistance += delta
+            state.layoutResult?.let { layoutResult ->
+                currentDragPosition = dragBeginOffset + dragTotalDistance
+                TextFieldDelegate.setCursorOffset(
+                    currentDragPosition!!,
+                    layoutResult,
+                    currentState.processor,
+                    currentOffsetMapping,
+                    currentState.onValueChange
+                )
+            }
+        }
+
+        override fun onStop() {
+            currentState.showFloatingToolbar = true
```

Comment:
Now we have a cuncurrent ownership with `TextFieldSelectionManager` of writing to `showFloatingToolbar`. `TextFieldSelectionManager` can change it to `false`, and we'll reset it to `false`.

If we need to disable the toolbar during drag, we need to introduce 2 variables, and write a combined value to `showFloatingToolbar`

Reply:
I did this similarly to other `TextDragObserver` objects. I tested it more thoroughly and decided that we don't need to handle the state of `showFloatingToolbar` here at all.
Deleted this from here.